### PR TITLE
Implement Gradual Ability Boosts Variant Rule

### PR DESCRIPTION
**Pull request by CarlosFdez**
image

### CODE REVIEW

File: src/module/system/settings/variant-rules.ts [hunk 1]
```diff
@@ -23,23 +29,20 @@ const SETTINGS: Record<string, SettingRegistration> = {
         name: "PF2E.SETTINGS.Variant.FreeArchetype.Name",
         hint: "PF2E.SETTINGS.Variant.FreeArchetype.Hint",
         scope: "world",
-        config: false,
         default: 0,
         type: Boolean,
```
Comment (stwlam) [inline src/module/system/settings/variant-rules.ts:33]: This is old but sure is confusing.

File: src/module/system/settings/variant-rules.ts [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+        hint: "PF2E.SETTINGS.Variant.AbilityScore.GradualBoosts.Hint",
+        scope: "world",
+        default: false,
+        type: Boolean,
+    },
     staminaVariant: {
         name: "PF2E.SETTINGS.Variant.Stamina.Name",
         hint: "PF2E.SETTINGS.Variant.Stamina.Hint",
         scope: "world",
-        config: false,
```
Comment (stwlam) [inline src/module/system/settings/variant-rules.ts:6]: Why remove this and not also `scope`?

Comment (CarlosFdez) [reply src/module/system/settings/variant-rules.ts:6]: that's a good idea! Let me do that too.

File: src/module/system/settings/variant-rules.ts [hunk 3]
```diff
@@ -127,6 +111,8 @@ export class VariantRulesSettings extends FormApplication {
 
     static registerSettings(): void {
         for (const [k, v] of Object.entries(SETTINGS)) {
+            v.config ??= false;
+            v.scope ??= "world";
```
Comment (stwlam) [inline src/module/system/settings/variant-rules.ts:115]:
```suggestion
            v.config = false;
            v.scope = "world";
```

File: static/lang/en.json [hunk 4]
```diff
@@ -2109,6 +2109,7 @@
         "Level18": "Level 18",
         "Level19": "Level 19",
         "Level20": "Level 20",
+        "LevelRange": "Level {minLevel} - {level}",
```
Comment (stwlam) [inline static/lang/en.json:2112]:
```suggestion
        "LevelRange": "Levels {minLevel} - {level}",
```

File: static/templates/system/settings/variant-rules-settings.html [hunk 5]
```diff
@@ -1,111 +1,125 @@
...
-    <p class="notes">{{localize 'PF2E.SETTINGS.Variant.MasterModifier.Name'}}</p>
-  </div>
-  <div class="form-group">
-    <label>{{localize 'PF2E.SETTINGS.Variant.LegendaryModifier.Name'}}</label>
-    <div class="form-fields">
-        <input type="number" name="proficiencyLegendaryModifier" value="{{proficiencyLegendaryModifier.value}}" data-dtype="Number">
+    <div class="form-group">
+        <label>{{localize "PF2E.SETTINGS.Variant.LegendaryModifier.Name"}}</label>
+        <div class="form-fields">
+            <input type="number" name="proficiencyLegendaryModifier" value="{{proficiencyLegendaryModifier.value}}" data-dtype="Number">
```
Comment (stwlam) [inline static/templates/system/settings/variant-rules-settings.html:113]:
```suggestion
            <input type="number" name="proficiencyLegendaryModifier" value="{{proficiencyLegendaryModifier.value}}" />
```

File: static/templates/system/settings/variant-rules-settings.html [hunk 6]
```diff
@@ -1,111 +1,125 @@
...
-    <p class="notes">{{localize 'PF2E.SETTINGS.Variant.ExpertModifier.Name'}}</p>
-  </div>
-  <div class="form-group">
-    <label>{{localize 'PF2E.SETTINGS.Variant.MasterModifier.Name'}}</label>
-    <div class="form-fields">
-        <input type="number" name="proficiencyMasterModifier" value="{{proficiencyMasterModifier.value}}" data-dtype="Number">
+    <div class="form-group">
+        <label>{{localize "PF2E.SETTINGS.Variant.MasterModifier.Name"}}</label>
+        <div class="form-fields">
+            <input type="number" name="proficiencyMasterModifier" value="{{proficiencyMasterModifier.value}}" data-dtype="Number">
```
Comment (stwlam) [inline static/templates/system/settings/variant-rules-settings.html:106]:
```suggestion
            <input type="number" name="proficiencyMasterModifier" value="{{proficiencyMasterModifier.value}}" />
```

File: static/templates/system/settings/variant-rules-settings.html [hunk 7]
```diff
@@ -1,111 +1,125 @@
...
-    <p class="notes">{{localize 'PF2E.SETTINGS.Variant.TrainedModifier.Name'}}</p>
-  </div>
-  <div class="form-group">
-    <label>{{localize 'PF2E.SETTINGS.Variant.ExpertModifier.Name'}}</label>
-    <div class="form-fields">
-        <input type="number" name="proficiencyExpertModifier" value="{{proficiencyExpertModifier.value}}" data-dtype="Number">
+    <div class="form-group">
+        <label>{{localize "PF2E.SETTINGS.Variant.ExpertModifier.Name"}}</label>
+        <div class="form-fields">
+            <input type="number" name="proficiencyExpertModifier" value="{{proficiencyExpertModifier.value}}" data-dtype="Number">
```
Comment (stwlam) [inline static/templates/system/settings/variant-rules-settings.html:99]:
```suggestion
            <input type="number" name="proficiencyExpertModifier" value="{{proficiencyExpertModifier.value}}" />
```

File: static/templates/system/settings/variant-rules-settings.html [hunk 8]
```diff
@@ -1,111 +1,125 @@
...
+        <label>{{localize "PF2E.SETTINGS.Variant.UntrainedModifier.Name"}}</label>
+        <div class="form-fields">
+            <input type="number" name="proficiencyUntrainedModifier" value="{{proficiencyUntrainedModifier.value}}" data-dtype="Number">
+        </div>
+        <p class="notes">{{localize "PF2E.SETTINGS.Variant.UntrainedModifier.Hint"}}</p>
+    </div>
+    <div class="form-group">
+        <label>{{localize "PF2E.SETTINGS.Variant.TrainedModifier.Name"}}</label>
+        <div class="form-fields">
+            <input type="number" name="proficiencyTrainedModifier" value="{{proficiencyTrainedModifier.value}}" data-dtype="Number">
```
Comment (stwlam) [inline static/templates/system/settings/variant-rules-settings.html:92]:
```suggestion
            <input type="number" name="proficiencyTrainedModifier" value="{{proficiencyTrainedModifier.value}}" />
```

File: static/templates/system/settings/variant-rules-settings.html [hunk 9]
```diff
@@ -1,111 +1,125 @@
...
+                {{/each}}
+                {{/select}}
+            </select>
+        </div>
+        <p class="notes">{{localize "PF2E.SETTINGS.Variant.Proficiency.Hint"}}</p>
+    </div>
+    <div class="form-group">
+        <label>{{localize "PF2E.SETTINGS.Variant.UntrainedModifier.Name"}}</label>
+        <div class="form-fields">
+            <input type="number" name="proficiencyUntrainedModifier" value="{{proficiencyUntrainedModifier.value}}" data-dtype="Number">
```
Comment (stwlam) [inline static/templates/system/settings/variant-rules-settings.html:85]:
```suggestion
            <input type="number" name="proficiencyUntrainedModifier" value="{{proficiencyUntrainedModifier.value}}" />
```

**Comment by CarlosFdez** [comment]
I had only changed the indentation, but sure. Remember to strip out all the sub-commits!

**Comment by stwlam** [comment]
> I had only changed the indentation, but sure. Remember to strip out all the sub-commits!

But then I don't get to be a 7x co-author!

**Pull request merged**